backend/tracim_backend/applications/collaborative_document_edition/controller.py

```diff
@@ -102,28 +102,37 @@ def create_file_from_template(
         collaborative_document_edition_api = CollaborativeDocumentEditionFactory().get_lib(
             current_user=request.current_user, session=request.dbsession, config=app_config
         )
-        collaborative_document_edition_api.check_template_available(hapic_data.body.template)
+
+        content = None # type: typing.Optional['Content']
         parent = None  # type: typing.Optional['Content']
-        if hapic_data.body.parent_id:
-            try:
-                parent = api.get_one(
-                    content_id=hapic_data.body.parent_id, content_type=content_type_list.Any_SLUG
-                )
-            except ContentNotFound as exc:
-                raise ParentNotFound(
-                    "Parent with content_id {} not found".format(hapic_data.body.parent_id)
-                ) from exc
-        content = api.create(
-            filename=hapic_data.body.filename,
-            content_type_slug=FILE_TYPE,
-            workspace=request.current_workspace,
-            parent=parent,
-        )
-        api.save(content, ActionDescription.CREATION)
-        with new_revision(session=request.dbsession, tm=transaction.manager, content=content):
-            collaborative_document_edition_api.update_content_from_template(
-                content=content, template_filename=hapic_data.body.template
+
+        if hapic_data.body.template_id:
+            template = api.get_one(hapic_data.body.template_id)
+            content = api.copy(template, new_label=hapic_data.body.filename, copy_revision=False)
+        else:
+            collaborative_document_edition_api.check_template_available(hapic_data.body.template)
+            if hapic_data.body.parent_id:
```

Comment:
I think you should still set the proper parent when a template_id is given, isn't it?

Reply:
Not really, if you are on space A and you want to use a template from space B, you want to your file to still be child of space A. This way you can sort every templates and put them into one folder while using them without having to change their position/parent

Reply:
I mean that the API allows to give a `parent_id` (`hapic_data.body.parent_id`). So I fail to see why it should not be used when using a template/copying from an existing content.

Reply:
Solved by using `copy` in `create`